industrial_ci/src/tests/source_tests.sh

```diff
@@ -79,8 +79,17 @@ rosdep --version
 if ! [ -d /etc/ros/rosdep/sources.list.d ]; then
     sudo rosdep init
 fi
-ret_rosdep=1
-rosdep update || while [ $ret_rosdep != 0 ]; do sleep 1; rosdep update && ret_rosdep=0 || echo "rosdep update failed"; done
+
+update_opts=()
+case "$ROS_DISTRO" in
+"hydro"|"jade")
+    if rosdep update --help | grep -q -- --include-eol-distros; then
+      update_opts+=(--include-eol-distros)
+    fi
+    ;;
+esac
+
+ici_retry 2 rosdep update "${update_opts[@]}"
```

Comment:
Just curious, what happens without retrying here?

Reply:
The retry was in there before..

Reply:
There is a change though: now a failed update will stop the test and not just print a warning..